app/uk/gov/hmrc/tariffclassificationfrontend/controllers/RequestActions.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.controllers
+
+import javax.inject.{Inject, Singleton}
+import play.api.mvc.{ActionBuilder, ActionRefiner}
+import uk.gov.hmrc.tariffclassificationfrontend.models.request.{AuthenticatedCaseRequest, AuthenticatedRequest}
+
+@Singleton
+class RequestActions @Inject()(authoriseFilter: AuthoriseFilterAction,
+                               checkPermissionsAction: CheckPermissionsAction,
+                               authenticatedAction: AuthenticatedAction,
+                               caseExistsActionFactory: VerifyCaseExistsActionFactory) {
+
+  def caseExistsAndFilterByAuthorisation(reference: String): ActionBuilder[AuthenticatedCaseRequest] = authenticatedAction andThen caseExists(reference) andThen authoriseFilter
```

Comment:
Personally, I think it would be easier to understand in the controllers if each one had to do 
`(verify.authorised andThen verify.caseExists(reference) andThen verify.userIsPermittedToModify(...))`
Or at least:
`(verify.authorisedAndCaseExists(reference) andThen verify.userIsPermittedToModify(...))`

Reply:
If we try and create "super" methods like this, they may save us some duplicate code, but they will cost us a lot of "time to understand whats going on"